execution-engine/contract_macro/tests/lib.rs

```diff
@@ -0,0 +1,31 @@
+use std::{
+    fs,
+    path::Path,
+    process::{Command, Stdio},
+};
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    #[test]
+    fn check_edge_cases() {
+        let t = trybuild::TestCases::new();
+        t.compile_fail("tests/templates/multiple_constructors.rs");
+        t.compile_fail("tests/templates/no_contract.rs");
+        t.compile_fail("tests/templates/no_constructor.rs");
+    }
+
+    #[test]
+    fn check_simple_output() {
+        let location = Path::new("../contracts/integration/sample_contract/src");
+        let expansion = Command::new("cargo")
+            .current_dir(location)
+            .arg("expand")
+            .stdout(Stdio::piped())
+            .output()
+            .expect("Failed to execute command");
```

Comment:
I'd suggest the following change to make debugging a failure simpler.  Holding the command in a variable means it can be printed on failure.

Also, it's not enough to just check that `Command::output` returns `Ok`.  That just means we were able to spawn a process afaik.  We need to check the resulting output's `status` to see if the actual command succeeded or not.

Finally, we don't need to specify `Stdio::piped()` if we're using `Command::output`, since that's its default (although also specifying it does no harm).
```suggestion

        let mut command = Command::new("cargo");
        command.current_dir(location).arg("expand");

        let expansion = command
            .output()
            .expect(&format!("failed to run '{:?}'", command));

        if !expansion.status.success() {
            let stdout = str::from_utf8(&expansion.stdout).expect("should be valid UTF8");
            let stderr = str::from_utf8(&expansion.stderr).expect("should be valid UTF8");
            panic!("failed to run '{:?}':\n{}\n{}\n", command, stdout, stderr);
        }

```